microdata_tools/_encrypt.py

```diff
@@ -58,24 +54,36 @@ def _encrypt_dataset(rsa_keys_dir: Path, dataset_dir: Path, output_dir: Path) ->
 
     dataset_output_dir = output_dir / dataset_name
     os.makedirs(dataset_output_dir)
+    os.makedirs(dataset_output_dir / "chunks", exist_ok=True)
 
-    encrypted_file = dataset_output_dir / f"{dataset_name}.csv.encr"
     encrypted_symkey_file = dataset_output_dir / f"{dataset_name}.symkey.encr"
 
     # Generate and store symmetric key for this file
     symkey = Fernet.generate_key()
 
     # Encrypt csv file
+    chunk_count = 0
+    logger.debug(f"Chunk size: {CHUNK_SIZE_BYTES} Bytes")
+
     with open(csv_file, "rb") as file:
-        data = file.read()  # Read the bytes of the input file
+        while True:
+            data = file.read(CHUNK_SIZE_BYTES)
+            if not data:
+                break
 
-    fernet = Fernet(symkey)
-    encrypted = fernet.encrypt(data)
+            chunk_count += 1
+            fernet = Fernet(symkey)
+            encrypted = fernet.encrypt(data)
 
-    with open(encrypted_file, "wb") as file:
-        file.write(encrypted)
+            chunk_file = (
+                dataset_output_dir
+                / "chunks"
+                / f"{dataset_name}_chunk_{chunk_count}.csv.encr"
```

Comment:
This file name might be unnecessarily verbose and causes us to have to use a regex later.
This file is already in the dataset output dir under chunks so if the filename is simply:
"chunk_{chunk_count"} we can simply get the file number by doing a [6:] slice

[ref comment on _get_sorted_files]

Reply:
Yes this is a good idea! Now we only take the contents of "chunk" dir when we package_dataset and add them to the root of the tar file. But I would perhaps be easier to just add the whole chunk dir to the tar.

Reply:
Oh I missed that the .tar was flat, I see that now 👀 But I like that! :100: